src/plugins/midi-piano/piano-component.js

```diff
@@ -0,0 +1,115 @@
+import PropTypes from 'prop-types';
+import { Piano } from 'react-piano';
+import React, { useEffect, useState, useRef } from 'react';
+
+export default function PianoComponent(props) {
+
+  const defaultKeyWidth = 29.2;
+  const renderCount = useRef(0);
+  const renderPiano = useRef(false);
+  const containerDiv = useRef(null);
+  const containerWidth = useRef(null);
+  const pianoWrapperDiv = useRef(null);
+  const pianoWrapperWidth = useRef(null);
+  const { noteRange, playNote, stopNote, keys } = props;
+  const [pianoWrapperDimensions, setPianoWrapperDimensions] = useState({});
+
+  function getNumberOfKeysRendered() {
+    return noteRange.last - noteRange.first + 1;
+  }
+
+  const numberOfKeysRendered = getNumberOfKeysRendered();
+
+  const getPianoWrapperDimensions = clientWidth => {
+    let width;
+    let height;
+    if (numberOfKeysRendered > 39) {
+      width = clientWidth;
+      const keyWidth = clientWidth / numberOfKeysRendered;
+      height = keyWidth * 5.5;
+      pianoWrapperWidth.current = width;
+      return { width: `${width}px`, height: `${height}px` };
+    }
+    const neededWidth = numberOfKeysRendered * defaultKeyWidth;
+    if (neededWidth > clientWidth) {
+      width = clientWidth;
+      const keyWidth = clientWidth / numberOfKeysRendered;
+      height = keyWidth * 5.5;
+    } else {
+      height = 160;
+      width = numberOfKeysRendered * defaultKeyWidth;
+    }
+    pianoWrapperWidth.current = width;
+    return { width: `${width}px`, height: `${height}px` };
+  };
+
+  const handleWindowResize = () => {
+    const neededWidth = numberOfKeysRendered * defaultKeyWidth;
+    if (containerDiv.current.clientWidth === containerWidth.current) {
+      return;
+    }
+    if (containerDiv.current.clientWidth < containerWidth.current && containerDiv.current.clientWidth > pianoWrapperWidth.current) {
+      return;
+    }
+    if (containerDiv.current.clientWidth > containerWidth.current && neededWidth < containerDiv.current.clientWidth) {
+      return;
+    }
+    const obj = getPianoWrapperDimensions(containerDiv.current.clientWidth);
+    setPianoWrapperDimensions(obj);
+  };
+
+  useEffect(() => {
+    renderCount.current += 1;
```

Comment:
Hmm, this looks like a workaround to try and delay executing the code from line 86 onward, right? 
Or what was the situation you were trying to get around of by doing this?